stone/target/python_rsrc/stone_serializers.py

```diff
@@ -24,11 +24,471 @@
 except (SystemError, ValueError):
     # Catch errors raised when importing a relative module when not in a package.
     # This makes testing this file directly (outside of a package) easier.
-    import stone_validators as bv  # type: ignore
+    import stone_validators as bv # type: ignore
+
+#-------------------------------------------------------------------------
+class StoneEncoderInterface(object):
+    """
+    Interface defining a stone object encoder.
+    """
+
+    def encode(self, validator, value):
+        """
+        Validate ``value`` using ``validator and return the encoding.
+
+        Args:
+            validator: the stone validator used to validate ``value``
+            value: the object to encode
+
+        Returns:
+            This is implementation-defined. The convention is to return an
+            encoded object where ``out_file`` is ``None``, and return
+            ``None`` if ``out_file`` is provided (and supported)
+
+        Raises:
+            stone_validators.ValidationError: Raised if ``value`` (or one
+                of its sub-values).
+        """
+        raise NotImplementedError
+
+    def encode_to_file(self, validator, value, out_file):
```

Comment:
Any reason why we're adding `encode_to_file`? We're not using it in server code right?

I can imagine a future where we do the encoding in a streaming fashion which would be cool, but this isn't that. It's just a json_dump wrapper which I think is unnecessary. The caller can do that themselves with the response from `encode()`.